client/reducers/reducer.js

```diff
@@ -91,6 +93,20 @@ const reducer = (state = INITIAL_STATE, action) => {
       newState.commandQueue.push(action.payload)
       return newState
 
+    // Has the command queue finished running? i.e. executed all commands
+    case 'HAS_FINISHED':
+    console.log('finished at reducer')
+      newState.hasFinished = true
+      return newState
+
+    case 'RETRY':
```

Comment:
Use the STOP_BUTTON action type instead. Just add newState.hasFinished = false to it